### PR TITLE
fix kube-proxy manifest

### LINKED ISSUE #80434: [Failing Test] Errors bringing up kube-proxy in CI
**Which jobs are failing**:
ci-kubernetes-e2e-gci-gce-alpha-features

**Which test(s) are failing**:
BeforeSuite 

**Since when has it been failing**:
Failing since 7/18 
The failure seems to have been introduced in https://github.com/kubernetes/kubernetes/compare/477304cdb...3f1cb97f9
https://github.com/kubernetes/kubernetes/pull/80229 is the only PR within that window that modifies any cluster configuration. This PR is related to https://github.com/kubernetes/kubernetes/issues/80262

**Testgrid link**:
https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-alpha-features

**Reason for failure**:
kube-proxy is stuck in a pending state

/milestone v1.16
/priority critical-urgent
/kind failing-test
/sig testing
/sig cluster-lifecycle
/cc @Verolop @jimangel @soggiest @alenkacz

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation

/kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: Should fix https://github.com/kubernetes/kubernetes/issues/80434 -- corrects `cluster/addons/kube-proxy/kube-proxy-ds.yaml` to match `cluster/saltbase/salt/kube-proxy/kube-proxy.manifest` and adds a note about keeping these manifests in sync in the other direction (`cluster/saltbase/salt/kube-proxy/kube-proxy.manifest` already had one).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/gce/manifests/kube-proxy.manifest [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# Please keep kube-proxy configuration in-sync with:
+# cluster/addons/kube-proxy/kube-proxy-ds.yaml
```
Comment (BenTheElder) [inline cluster/gce/manifests/kube-proxy.manifest]: see lines 1-2 of `cluster/addons/kube-proxy/kube-proxy-ds.yaml`

https://github.com/kubernetes/kubernetes/pull/80566/files#diff-d83ae2fa2959cf05c2e8d4f4a5ac2601R1

**Comment by hoegaarden** [comment]
This is needed because of the changes in https://github.com/kubernetes/kubernetes/issues/80262 / https://github.com/kubernetes/kubernetes/pull/80229. It is related to https://github.com/kubernetes/kubernetes/pull/80229, which switched the image name for other parts of `./cluster/...`

/lgtm

**Comment by neolit123** [comment]
seems correct as per similar manifests.
ideally these should be using manifest lists even if targeting CI only at this point.

LGTM.

**Comment by alejandrox1** [comment]
For final approval 
/assign @MrHohn

**Comment by BenTheElder** [comment]
> ideally these should be using manifest lists even if targeting CI only at this point.

Yeah, the problem is we only generate manifests during a full release process, apparently in the release repo. We'd have to get that ported over first and fixed to work with quick releases. 

That will need more discussion, it seems for now CI is just referencing the amd64 images.

**Review by MrHohn — APPROVED** [review]
Thanks for the fix!
/lgtm
/approve

**Comment by justaugustus** [comment]
Thanks @BenTheElder!
/lgtm
/approve

**Comment by justaugustus** [comment]
(kicking the bots)
/hold

**Pull request merged**